src/manifest.c

```diff
@@ -77,6 +77,27 @@ int file_sort_filename_reverse(const void *a, const void *b)
 	return -ret;
 }
 
+int file_sort_hash(const void *a, const void *b)
+{
+	struct file *A, *B;
+	int ret;
+	A = (struct file *)a;
+	B = (struct file *)b;
+
+	ret = strcmp(A->hash, B->hash);
+	if (ret) {
+		return ret;
+	}
+	if (A->is_deleted > B->is_deleted) {
```

Comment:
we just add files to the list if if_deleted == 0. Why do we need this here?

Reply:
we don't, I copied this function from `file_sort_filename` and forgot to remove that part